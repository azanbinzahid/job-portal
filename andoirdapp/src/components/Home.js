import React from 'react';
import {connect} from 'react-redux';
import {View, Text, Button} from 'react-native';

const Home = (props) => {
  return (
    <>
      <View style={{flex: 1}}>
        <View style={{flex: 1}}>
          <Text> Welcome {props.firstName}</Text>
          <Text>
            Looking for a job? Apply online for latest jobs in Pakistan. Browse
            vacancies and apply for the latest jobs near you.
          </Text>
          {props.isLogged ? (
            <Button title="Explore Jobs" />
          ) : (
            <Button title="Login" />
          )}
        </View>
      </View>
    </>
  );
};

const mapStateToProps = (state) => ({
  firstName: state.userReducer.user.firstName,
  isLogged: state.userReducer.loggedIn,
  jobCount: state.jobReducer.jobs.length,
});

export default connect(mapStateToProps)(Home);
