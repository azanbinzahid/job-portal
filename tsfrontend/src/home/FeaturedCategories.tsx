import React, { FC } from "react";
import { connect } from "react-redux";
import { Container, Button, Col, Row } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";
import { RootState } from "redux/reducers";
import { FaAirbnb } from "react-icons/fa";
import SectionHeading from "app/common/SectionHeading";

type Props = {
  filters: any;
};

export const FeaturedCategories: FC<Props> = ({ filters }) => {
  return (
    <Container<React.ElementType> className="" align="center">
      <SectionHeading title="Featured Categories" />

      <Row>
        {filters.Category
          ? filters.Category.map((e: any) => {
              return (
                <Col className="pb-3">
                  <LinkContainer to={`/jobs/?&category=${e.name}`}>
                    <Button variant="link" block>
                      <h1>
                        <FaAirbnb />
                      </h1>
                      {e.name}
                    </Button>
                  </LinkContainer>
                </Col>
              );
            })
          : null}
      </Row>
    </Container>
  );
};

const mapStateToProps = (state: RootState) => ({
  filters: state.jobReducer.filters,
});

export default connect(mapStateToProps)(FeaturedCategories);
