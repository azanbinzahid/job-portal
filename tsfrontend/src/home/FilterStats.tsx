import React, { FC } from "react";
import { connect } from "react-redux";
import { Container, Jumbotron, Button, Col, Row } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";
import { RootState } from "redux/reducers";
import { pluralize } from "utils/helper";

type Props = {
  filters: any;
};

export const FilterStats: FC<Props> = ({ filters }) => {
  return (
    <Jumbotron<React.ElementType> className="pt-5 pb-5" align="center">
      <h1 className="pb-5">Easily filter jobs across</h1>
      <Container>
        <Row>
          {Object.keys(filters).map((key) => {
            return (
              <Col className="pb-3">
                <LinkContainer to="/jobs">
                  <Button variant="outline-dark">
                    <h1>{filters[key].length}+ </h1>
                    {pluralize(key)}
                  </Button>
                </LinkContainer>
              </Col>
            );
          })}
        </Row>
      </Container>
    </Jumbotron>
  );
};

const mapStateToProps = (state: RootState) => ({
  filters: state.jobReducer.filters,
});

export default connect(mapStateToProps)(FilterStats);
