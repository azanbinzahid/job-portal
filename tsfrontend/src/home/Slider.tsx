import Search from "jobs/Search";
import React, { FC } from "react";
import { Carousel } from "react-bootstrap";
import styled from "styled-components";

const BannerTitle = styled.h1`
  display: block;
  margin-bottom: 12px;
  font-size: 47px;
  font-family: "Jost", "PT Serif", sans-serif;
  font-weight: 700;
  color: #fff;
`;

const BannerText = styled.p`
  display: block;
  margin-bottom: 12px;
  font-family: "Jost", "PT Serif", sans-serif;
  font-weight: 700;
  color: #fff;
`;

const Image = styled.img`
  filter: brightness(50%);
`;

const Slider: FC = () => {
  return (
    <Carousel>
      <Carousel.Item>
        <Image
          // className="d-block w-100"
          src="https://www.jobs.punjab.gov.pk/new_design/images/banner2.jpg"
          alt="First slide"
        />
        <Carousel.Caption>
          <BannerTitle>Get The Career You Deserve</BannerTitle>
          <Search />
          <BannerText> For example: remote, lahore, manager </BannerText>
        </Carousel.Caption>
      </Carousel.Item>
    </Carousel>
  );
};

export default Slider;
