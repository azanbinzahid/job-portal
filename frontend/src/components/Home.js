import React from 'react'
import {useSelector} from 'react-redux'
import { Container, Jumbotron, Button, Badge} from 'react-bootstrap'
import { LinkContainer } from 'react-router-bootstrap';
import Slider from 'components/Slider'

export default function Home() {

    const userReducer = useSelector(state => state.userReducer)
    const jobReducer = useSelector(state => state.jobReducer)
    const jobCount = jobReducer.jobs.length

    return (
        <div>
            <Slider/>
            <Container className="p-5">
                <Jumbotron>
                    <h2> Welcome {userReducer.user.first_name}</h2>
                    <p>
                    Looking for a job? Apply online for latest jobs in Pakistan. Browse vacancies and apply for the latest jobs near you.
                    </p>
                    {
                        userReducer.loggedIn ? ( 
                        <LinkContainer to="/jobs">
                        <Button variant="primary">
                            Explore Jobs <Badge variant="light">{jobCount}</Badge>
                        </Button>
                        </LinkContainer>
                        ) : (
                            <LinkContainer to="/login">
                            <Button variant='primary'> Login </Button>
                            </LinkContainer>                        
                        )
                    }
                </Jumbotron>
            </Container>
        </div>
    )
}
