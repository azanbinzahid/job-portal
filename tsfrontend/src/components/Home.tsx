import React, { FC } from "react";
import { connect } from "react-redux";
import { Container, Jumbotron, Button, Badge } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";
import Slider from "components/Slider";
import { RootState } from "redux/reducers";

type Props = {
  firstName: string;
  jobCount: number;
  isLogged: boolean;
};

const Home: FC<Props> = ({ firstName, jobCount, isLogged }) => {
  return (
    <div>
      <Slider />
      <Container className="p-5">
        <Jumbotron>
          <h2> Welcome {firstName}</h2>
          <p>
            Looking for a job? Apply online for latest jobs in Pakistan. Browse
            vacancies and apply for the latest jobs near you.
          </p>
          {isLogged ? (
            <LinkContainer to="/jobs">
              <Button variant="primary">
                Explore Jobs <Badge variant="light">{jobCount}</Badge>
              </Button>
            </LinkContainer>
          ) : (
            <LinkContainer to="/login">
              <Button variant="primary"> Login </Button>
            </LinkContainer>
          )}
        </Jumbotron>
      </Container>
    </div>
  );
};

const mapStateToProps = (state: RootState) => ({
  firstName: state.userReducer.user.firstName,
  isLogged: state.userReducer.loggedIn,
  jobCount: state.jobReducer.jobs.length,
});

export default connect(mapStateToProps)(Home);
