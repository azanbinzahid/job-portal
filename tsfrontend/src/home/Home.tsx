import React, { FC } from "react";
import { connect } from "react-redux";
import { Container, Jumbotron, Button, Col, Row, Badge } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";
import Slider from "home/Slider";
import { RootState } from "redux/reducers";
import { FaBeer } from "react-icons/fa";

type Props = {
  firstName: string;
  isLogged: boolean;
  filters: any;
};

export const Home: FC<Props> = ({ firstName, isLogged, filters }) => {
  return (
    <>
      <Slider />
      <Container<React.ElementType> className="pt-5 pb-5" align="center">
        <Row>
          {Object.keys(filters).map((key) => {
            return (
              <Col>
                <LinkContainer to="/jobs">
                  <Button variant="outline-primary">
                    <h1>
                      {filters[key].length}
                      <Badge variant="warning"></Badge>
                    </h1>
                    {key}{" "}
                  </Button>
                </LinkContainer>
              </Col>
            );
          })}
        </Row>
      </Container>
    </>
  );
};

const mapStateToProps = (state: RootState) => ({
  firstName: state.userReducer.user.firstName,
  isLogged: state.userReducer.loggedIn,
  filters: state.jobReducer.filters,
});

export default connect(mapStateToProps)(Home);
