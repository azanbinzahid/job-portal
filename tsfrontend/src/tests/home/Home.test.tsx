import React from "react";
import { shallow } from "enzyme";
import { Home } from "home/Home";
test("renders the component", () => {
  const component = shallow(
    <Home firstName="name" jobCount={100} isLogged={true} />
  );
  expect(component).toMatchSnapshot();
});

test("renders the component", () => {
  const component = shallow(
    <Home firstName="name" jobCount={100} isLogged={false} />
  );
  expect(component).toMatchSnapshot();
});
