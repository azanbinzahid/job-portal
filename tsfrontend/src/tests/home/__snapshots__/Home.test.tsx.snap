// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Fragment>
  <Slider />
  <Connect(Component) />
  <Connect(Component) />
  <SectionHeading
    title="Popular Jobs"
  />
  <Connect(Component)
    limit={3}
  />
</Fragment>
`;
